src/components/Header/SearchBar/SearchBar.tsx

```diff
@@ -38,21 +37,21 @@ const SearchBar: React.FC = () => {
   const [debouncedValue] = useDebounce(inputValue, 500);
 
   useEffect(() => {
-    setInputValue(selected?.properties.NM_MUN ?? '');
-  }, [selected, setInputValue]);
+    const hasDebouncedValue = !!debouncedValue;
 
-  useEffect(() => {
-    const query = debouncedValue
-      .toLowerCase()
-      .replace(/\s+/g, ' ')
-      .replace(/(^\s+|\s+$)/g, '');
+    if (hasDebouncedValue) {
+      const query = debouncedValue
+        .toLowerCase()
+        .replace(/\s+/g, ' ')
+        .replace(/(^\s+|\s+$)/g, '');
 
-    const hasQuery = query !== '';
+      const hasQuery = query !== '';
 
-    if (hasQuery) {
-      setDistrictSearched(getSortedDistricts(getFilteredDistricts(all, query)));
-    } else {
-      setDistrictSearched(getSortedDistricts(all));
+      if (hasQuery) {
+        setDistrictSearched(getSortedDistricts(getFilteredDistricts(all, query)));
```

Comment:
**nitpick:** can you please add a variable explaining what is this before set as value in the state

```suggestion
        const variable = getSortedDistricts(getFilteredDistricts(all, query))
        setDistrictSearched(variable);
```